substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/JavaMonitorEnterEvent.java

```diff
@@ -0,0 +1,48 @@
+package src.com.oracle.svm.core.src.com.oracle.svm.core.jfr.events;
+
+
+import org.graalvm.nativeimage.StackValue;
+import org.graalvm.word.UnsignedWord;
+
+import com.oracle.svm.core.annotate.Uninterruptible;
+import com.oracle.svm.core.jfr.JfrEvent;
+import com.oracle.svm.core.jfr.JfrNativeEventWriter;
+import com.oracle.svm.core.jfr.JfrNativeEventWriterData;
+import com.oracle.svm.core.jfr.JfrNativeEventWriterDataAccess;
+import com.oracle.svm.core.jfr.JfrTicks;
+import com.oracle.svm.core.jfr.SubstrateJVM;
+
+
+public class JavaMonitorEnterEvent {
+
+    @Uninterruptible(reason = "Accesses a JFR buffer.")
+    public static void emit(Object obj, org.graalvm.nativeimage.IsolateThread previousOwner,long startTicks) {
+        emit(obj, com.oracle.svm.core.jfr.SubstrateJVM.get().getThreadId(previousOwner), startTicks);
+    }
+    @Uninterruptible(reason = "Accesses a JFR buffer.")
+    public static void emit(Object obj, long previousOwner, long startTicks) {
+        if (SubstrateJVM.isRecording() && SubstrateJVM.get().isEnabled(JfrEvent.JavaMonitorEnter)) {
+            JfrNativeEventWriterData data = StackValue.get(JfrNativeEventWriterData.class);
+            JfrNativeEventWriterDataAccess.initializeThreadLocalNativeBuffer(data);
+
+            if (emit0(data, obj, previousOwner, startTicks, SubstrateJVM.get().isLarge(JfrEvent.JavaMonitorEnter)) == com.oracle.svm.core.jfr.JfrEventWriteStatus.RetryLarge) {
```

Comment:
There shouldn't be any need for the retry mechanism as this event has a fixed size (it only contains primitive data). Search the code base for `JfrNativeEventWriter.beginSmallEvent(...)` to find examples where we emit JFR events that have a fixed size.

Reply:
Please also use your Red Hat email address for all commits. Otherwise, we can't merge the changes to master easily.